tests/ui/coherence/coherence-overlap-negative-impls.rs

```diff
@@ -39,6 +36,7 @@ mod private {
     pub auto trait NotNil {}
 
     impl !NotNil for Is<Nil> {}
+    //~^ ERROR `!NotNil` impls cannot be specialized
 }
```

Comment:
I dislike this test and would like it removed. #74629 has been fixed by #113312. With the removal of non-always applicable negative impls it doesn't really test anyhting related to the original issue anymore.